include/swift/SIL/SILInstruction.h

```diff
@@ -1222,12 +1222,14 @@ class StringLiteralInst final : public LiteralInst,
 private:
   unsigned Length;
   Encoding TheEncoding;
+  bool NullTerminated;
 
   StringLiteralInst(SILDebugLocation *DebugLoc, StringRef text,
-                    Encoding encoding, SILType ty);
+                    Encoding encoding, bool nullTerminate, SILType ty);
```

Comment:
Can you be specific as to what breaks? You could still avoid the caller having to copy by having the constructor take a `Twine` instead, maybe, or take the flag and then append the null terminator in the `create`-d buffer that gets allocated.